### PR TITLE
`AdtDef` is an algebraic data type, not abstract data type

### PR DESCRIPTION
**Pull request by Centril**
r? @varkor

### CODE REVIEW

File: src/librustc/ty/mod.rs [hunk 1]
```diff
@@ -1938,7 +1938,11 @@ pub struct FieldDef {
     pub vis: Visibility,
 }
 
-/// The definition of an abstract data type -- a struct or enum.
+/// The definition of an [*algebraic data type (ADT)*][adt].
+/// These include `struct`s (products), `enum` (sums).
```
Comment (Centril) [inline src/librustc/ty/mod.rs]:
```suggestion
/// These include `struct`s (products) and `enum`s (sums).
```

**Comment by varkor** [comment]
I think both "abstract data type" and "algebraic data type" are incorrect. "Abstract" because it's actually concrete and "algebraic" because it includes unions and is also not as general as the usual sense of "algebraic data type". I think @eddyb wasn't happy with the existing name as well; maybe `UserDataType` is a better choice.

File: src/librustc/ty/mod.rs [hunk 2]
```diff
@@ -1938,7 +1938,11 @@ pub struct FieldDef {
     pub vis: Visibility,
 }
 
-/// The definition of an abstract data type -- a struct or enum.
+/// The definition of an [*algebraic data type (ADT)*][adt].
```
Comment (varkor) [inline src/librustc/ty/mod.rs]:
```suggestion
/// The definition of a user-defined type, such as a `struct`, `enum` or `union`.
```

Comment (varkor) [reply src/librustc/ty/mod.rs]:
```rust
/// "ADT" stands for "A data type".
```
😎

Comment (eternaleye) [reply src/librustc/ty/mod.rs]: Also available is "Arbitrary Data Type"

**Comment by Centril** [comment]
@varkor Adjusted to "user-defined..." but also included a note at the bottom re. algebraic data types and how it is slightly wrong.

**Pull request merged**